Sources/ParseSwift/Authentication/3rd Party/ParseFacebook.swift

```diff
@@ -0,0 +1,395 @@
+
+    /// Authentication keys required for Facebook authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id // swiftlint:disable:this identifier_name
+        case authenticationToken
+        case accessToken
+        case expirationDate
+        
+        enum CodingKeys: String, CodingKey {
+          case id // swiftlint:disable:this identifier_name
+          case token
+          case accessToken = "access_token"
+          case expirationDate = "expiration_date"
+        }
+        /// Properly makes an authData dictionary with the required keys.
+        /// - parameter userId: Required id for the user.
+        /// - parameter authenticationToken: Required identity token for the user for Facebook limited login.
+        /// - parameter accessToken: Required identity token for the user for Facebook graph API
+        /// - parameter expirationDate: required expiration data for user authentication for Facebook login
+        /// - returns: Required authData dictionary.
+        func makeDictionary(userId: String?, accessToken: String?, authenticationToken: String?, expirationDate: Date?) throws -> [String: String]? {
+            
+            let validAuthenticationToken = authenticationToken != nil || accessToken != nil
+            guard let  userId = userId, let expirationDate = expirationDate else {
+                throw ParseError(code: .unknownError, message: "userId or expirationDate can't be empty")
+            }
+            let dateString = DateFormatter.dateFormatter.string(from: expirationDate)
+            
+            var returnDictionary = [AuthenticationKeys.id.rawValue: userId,
+                                    AuthenticationKeys.expirationDate.rawValue: dateString]
+            guard validAuthenticationToken else {
+                throw ParseError(code: .unknownError, message: "provide either accessToken or authenticationToken can't be empty")
+            }
+            
+            if let accessToken = accessToken {
+              returnDictionary[AuthenticationKeys.accessToken.rawValue] = accessToken
+            }  else if let authenticationToken = authenticationToken {
+              returnDictionary[AuthenticationKeys.authenticationToken.rawValue] = authenticationToken
+            }
+            return returnDictionary
+        }
+
+        /// Verifies all mandatory keys are in authData.
+        /// - parameter authData: Dictionary containing key/values.
+        /// - returns: `true` if all the mandatory keys are present, `false` otherwise.
+        func verifyMandatoryKeys(authData: [String: String]?) -> Bool {
+            guard let authData = authData,
+                  authData[AuthenticationKeys.id.rawValue] != nil,
+                  authData[AuthenticationKeys.expirationDate.rawValue] != nil else {
+                return false
+            }
+            
+            if let _ = authData[AuthenticationKeys.accessToken.rawValue] {
```

Comment:
Can you check the warnings Swiftlint is giving you in Xcode? The way these lines are written and the spacing, I’m sure there’s a number of warnings that need to be addressed. There should only be 1 warning in the SDK and it has a comment saying to leave it

Reply:
I resolved this